packages/transition-backend/src/services/transitRouting/TrRoutingBatch.ts

```diff
@@ -317,26 +309,26 @@ class TrRoutingBatch {
         return { odTrips, errors };
     };
 
-    private startTrRoutingInstances = async (odTripsCount: number): Promise<number> => {
+    private startTrRoutingInstances = async (odTripsCount: number): Promise<[number, number]> => {
         // Divide odTripCount by 3 for the minimum number of calculation, to avoid creating too many processes if trip count is small
         const trRoutingInstancesCount = Math.max(
             1,
             Math.min(Math.ceil(odTripsCount / 3), this.transitRoutingAttributes.cpuCount || 1)
         );
-
+        let trRoutingPort = -1;
         try {
             this.options.progressEmitter.emit('progress', { name: 'StartingRoutingParallelServers', progress: 0.0 });
 
             // Because of cancellation, we need to make sure processes are stopped before restarting
             // TODO trRouting should be multi-threaded, this will be useless then.
-            await TrRoutingProcessManager.stopMultiple(trRoutingInstancesCount);
-            const startStatus = await TrRoutingProcessManager.startMultiple(trRoutingInstancesCount);
-
+            await TrRoutingProcessManager.stopBatch();
+            const startStatus = await TrRoutingProcessManager.startBatch(trRoutingInstancesCount);
+            trRoutingPort = startStatus.port;
             console.log('trRouting multiple startStatus', startStatus);
         } finally {
             this.options.progressEmitter.emit('progress', { name: 'StartingRoutingParallelServers', progress: 1.0 });
         }
-        return trRoutingInstancesCount;
+        return [trRoutingInstancesCount, trRoutingPort];
```

Comment:
You can put the return in the try, to avoid adding the initializing the trRoutingPort above and giving the impression that -1 might be a possible value. There is no catch for a potential error anyway.

Reply:
oh I see your comment about the no catch. Having just the finally was a bit confusing. Will need to have a comment about the no catch, just in case somebody add one someday.